examples/knative/powercli/kn-pcli-tag/handler.ps1

```diff
@@ -0,0 +1,64 @@
+Function Process-Handler {
+   param(
+      [Parameter(Position=0,Mandatory=$true)][CloudNative.CloudEvents.CloudEvent]$CloudEvent
+   )
+
+   # Decode CloudEvent
+   $cloudEventData = $cloudEvent | Read-CloudEventJsonData -ErrorAction SilentlyContinue -Depth 10
+   if($cloudEventData -eq $null) {
+      $cloudEventData = $cloudEvent | Read-CloudEventData
+   }
+
+   if(${env:FUNCTION_DEBUG} -eq "true") {
+      Write-Host "DEBUG: K8s Secrets:`n${env:TAG_SECRET}`n"
+
+      Write-Host "DEBUG: CloudEventData`n $(${cloudEventData} | Out-String)`n"
+   }
+
+   if(${env:TAG_SECRET}) {
+      $jsonSecrets = ${env:TAG_SECRET} | ConvertFrom-Json
+   } else {
+      Write-Host "K8s secrets `$env:TAG does not look to be defined"
+      break
+   }
+
+   # Extract all tag secrets for ease of use in function
+   $VCENTER_SERVER = ${jsonSecrets}.VCENTER_SERVER
+   $VCENTER_USERNAME = ${jsonSecrets}.VCENTER_USERNAME
+   $VCENTER_PASSWORD = ${jsonSecrets}.VCENTER_PASSWORD
+   $VCENTER_TAG_NAME = ${jsonSecrets}.VCENTER_TAG_NAME
+
+   # Extract VM Name from event
+   $vmName = $cloudEventData.Vm.Name
+
+   # Configure TLS 1.2/1.3 support as this is required for latest vSphere release
+   [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12 -bor [System.Net.SecurityProtocolType]::Tls13
+
+   Write-Host "$(Get-Date) - Configuring PowerCLI Configuration Settings`n"
+   try {
+      Set-PowerCLIConfiguration -InvalidCertificateAction:Ignore -ParticipateInCeip:$true -Confirm:$false -WarningVariable Warn | Out-Null
```

Comment:
What is this doing?

Reply:
Disables TLS certificate validation as this is needed for anyone with vCenter Server that uses self-sign TLS certificate. This was done for testing purposes but fair point that this should be flag in secret and this way it, can handle both scenarios